engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/multiinstance/MultiInstanceBodyCompletedHandler.java

```diff
@@ -112,7 +114,7 @@ private DirectBuffer insertAt(
 
     variableReader.wrap(array, 0, array.capacity());
     variableReader.readArrayHeader();
-    variableReader.skipValues(index - 1);
+    variableReader.skipValues((long) index - 1L);
```

Comment:
why we need a long here?

Reply:
I don't know. It's coming from the `MsgPackReader`.

Reply:
It was considered a bug by Sonar